.github/workflows/ci-scripts.yml

```diff
@@ -5,9 +5,33 @@ on:
   pull_request:
 
 jobs:
+  changes:
+    name: Find changed files
+    runs-on: ubuntu-latest
+    outputs:
+      not_ignored_changes: ${{ steps.filter.outputs.every_changed == 'true' || steps.filter.outputs.any_changed == 'true' }}
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      - name: Filter other changes
+        uses: ./.github/actions/filter-changes
+        id: filter
+        with:
+          # Filter files that match **every** pattern in this list
+          every-filter: |
+            - '**'                     # Initially include all files
+            - '!**/.github/**'         # Exclude all changes under the .github directory
+          # Filter files that match **any** of these patterns
+          any-filter: |
+            - '.github/workflows/ci-scripts.yml'
+            - '.github/actions/golangci-lint/**'
+            - '.github/actions/setup-go/**'
+            - '.github/actions/filter-changes/**'
+
   lint-scripts:
     # We don't directly merge dependabot PRs, so let's not waste the resources
-    if: ${{ (github.event_name == 'pull_request' ||  github.event_name == 'schedule') && github.actor != 'dependabot[bot]' }}
+    needs: changes
+    if: ${{ (github.event_name == 'pull_request' ||  github.event_name == 'schedule') && github.actor != 'dependabot[bot]' && needs.changes.outputs.not_ignored_changes == 'true' }}
```

Comment:
This doesn't look right to me. We expect scheduled runs to execute regardless of changes.